app/components/clipboard_button_component.rb

```diff
@@ -0,0 +1,14 @@
+class ClipboardButtonComponent < ButtonComponent
+  attr_reader :clipboard_text, :tag_options
+
+  def initialize(clipboard_text:, **tag_options)
+    super(**tag_options)
+
+    @clipboard_text = clipboard_text
+    @tag_options = tag_options
+  end
+
+  def call
+    content_tag(:'lg-clipboard-button', super, data: { clipboard_text: clipboard_text })
```

Comment:
One possible future awkwardness we may run into is that because we're just wrapping the base markup, things like `tag_options[:class]` will apply to the child element, not the root. A small difference, but personally I'd expect `tag_options` to apply to the root element.